### PR TITLE
Add Young2020epidemiologic

### PR DESCRIPTION
**Pull request by WenfeiXiao**
Data for the nasopharyngeal swab results were obtained from the combined dataset in the supplementary materials of Challenger et al. BMC Medicine (2022) 20:25 (https://doi.org/10.1186/s12916-021-02220-0). The standard curve was calculated based on the concentration provided by Goyal et al. (2020)(https://www.science.org/doi/10.1126/sciadv.abc7112).

### CODE REVIEW

File: data/young2020epidemiologic/young2020epidemiologic-extraction.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+```
+
+Raw data ([CombinedDataset](https://github.com/shedding-hub/shedding-hub/blob/main/data/young2020epidemiologic/CombinedDataset.xlsx)), which is stored on [Shedding Hub](https://github.com/shedding-hub), will be loaded and cleaned to match the most updated [schema](https://github.com/shedding-hub/shedding-hub/blob/main/data/.schema.yaml).
+
+```python
+df = pd.read_excel(os.path.join(current_directory, 'Challenger2022/CombinedDataset.xlsx'), sheet_name='Viral_Load')
+young2020 = df[df['StudyNum'] == 6].copy()
+columns_to_drop = ['Estimated', 'SevMax', 'Sev1st', 'Died', 'SevMax3']
+young2020 = young2020.drop(columns=columns_to_drop)
+# Transform Ct value to gc/swab using obtained standard curve
```
Comment (YWAN446) [inline data/young2020epidemiologic/young2020epidemiologic-extraction.md:22]: Consider adding some descriptions of how the standard curve was back-calculated and maybe adding a visualization (points with the standard curve from Goyal data, Viral_Loads.csv) here in the Markdown file. Write down the equation of the standard curve in a Markdown block and explain this is obtained from Goyal et al. (2020).

**Review by YWAN446 — COMMENTED** [review]
Great work, Fiona. Few minor comments. I will encourage you to write some description in the Markdown files to explain how we back-calculate the standard curve from Goyal paper (with a plot maybe) and how we used the standard curve to calculate the concentration (in gc/swab) using the CT values from Challenger paper. 

I will test the reproducibility after you update those.

File: data/young2020epidemiologic/young2020epidemiologic.yaml [hunk 2]
```diff
@@ -0,0 +1,694 @@
...
+  in Singapore
+doi: 10.1001/jama.2020.3204
+description: >
+  Clinical, laboratory, and radiologic data were collected, including PCR cycle threshold
+  values from nasopharyngeal swabs and viral shedding in blood, urine, and stool.
+  The clinical course was summarized, including the requirement for supplemental oxygen,
+  intensive care, and the use of empirical treatment with lopinavir-ritonavir. The
+  numbers of positive stool, blood, and urine samples were small. Data for the nasopharyngeal
+  swab results were obtained from the combined dataset in the supplementary materials
+  of Challenger et al. BMC Medicine (2022) 20:25 (https://doi.org/10.1186/s12916-021-02220-0).
```
[the remark this answers is not in the document]
Comment (WenfeiXiao) [reply data/young2020epidemiologic/young2020epidemiologic.yaml:12]: Yes, sorry. I'll correct this part and update description!

File: data/young2020epidemiologic/young2020epidemiologic-extraction.md (hunk 1 above)
Comment (WenfeiXiao) [reply data/young2020epidemiologic/young2020epidemiologic-extraction.md:22]: Just to confirm, if I want to add an image to a .md file, I can embed an image link using a format similar to how we include links for datasets. I can link to an image uploaded to the 'young2020epidemiologic' folder. Or is there a specific place where you would like me to store the image or a preferred naming convention for the image file, or is there a better way to include images? Please let me know 😃.

Comment (YWAN446) [reply data/young2020epidemiologic/young2020epidemiologic-extraction.md:22]: Yes, I think you need to include the figure in the  'young2020epidemiologic' folder and insert
`plot`
in your md code.

File: data/young2020epidemiologic/young2020epidemiologic-extraction.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+import yaml
+import pandas as pd
+import numpy as np
+from shedding_hub import folded_str, literal_str
+```
+
+Raw data ([CombinedDataset](https://github.com/shedding-hub/shedding-hub/blob/main/data/young2020epidemiologic/CombinedDataset.xlsx)), which is stored on [Shedding Hub](https://github.com/shedding-hub), will be loaded and cleaned to match the most updated [schema](https://github.com/shedding-hub/shedding-hub/blob/main/data/.schema.yaml).
+
+```python
+df = pd.read_excel(os.path.join(current_directory, 'Challenger2022/CombinedDataset.xlsx'), sheet_name='Viral_Load')
```
Comment (tillahoffmann) [inline data/young2020epidemiologic/young2020epidemiologic-extraction.md:18]:
```suggestion
df = pd.read_excel('CombinedDataset.xlsx', sheet_name='Viral_Load')
```

File: data/young2020epidemiologic/young2020epidemiologic-extraction.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+model.fit(X, Y)
+predictions = model.predict(X)
+slope = model.coef_[0]
+intercept = model.intercept_
+print(f"Slope: {slope}")
+print(f"Intercept: {intercept}")
+```
+![Linear Regression Plot](data/young2020epidemiologic/Goyal_linreg.png)
+
+Cycle threshold values were plotted against the corresponding viral load (VL) values per swab in ![Linear Regression Plot](data/young2020epidemiologic/Goyal_linreg.png). A linear regression was performed to obtain the slope and intercept of the standard curve and serves as the basis for converting `Ct` values from other samples in [Challenger et al. (2022)](https://doi.org/10.1186/s12916-021-02220-0) into viral load concentration.
```
Comment (tillahoffmann) [inline data/young2020epidemiologic/young2020epidemiologic-extraction.md:44]:
```suggestion
![Linear Regression Plot](Goyal_linreg.png)

Cycle threshold values were plotted against the corresponding viral load (VL) values per swab in the figure above. A linear regression was performed to obtain the slope and intercept of the standard curve and serves as the basis for converting `Ct` values from other samples in [Challenger et al. (2022)](https://doi.org/10.1186/s12916-021-02220-0) into viral load concentration.
```

**Review by tillahoffmann — DISMISSED** [review]
This looks great! Really nice to see that we can also extract nice data from more complex settings where only Ct values were reported. Feel free to merge whenever you're ready. 🚀

**Comment by YWAN446** [comment]
@tillahoffmann I made some additional changes when tested the reproducibility. However, I had to do `git pull` before `push` those changes as the remote branch has been updated. There were some conflicts as you made some changes yesterday, so I resolved those conflicts by manually comparing and editing as well as `accepting changes` from my current branch. Is the commit `1fca6e6` OK?
The main changes beyond what you did yesterday are: 1) changing the equation to the math mode; 2) changing the `-` to `+` in the calculation of `concentration` from `Ctvalue`; and 3) some values changed a tiny bit last two or three digits when reproducing the YAML.
If those changes look good to you, you can go ahead and close this PR.

**Comment by tillahoffmann** [comment]
Nice, thank you for fixing the sign. Do you know why the results didn't change by more given the sign change?

**Comment by YWAN446** [comment]
When I rerun the codes, the `-` gives incorrect results. Changing to `+` gives correct results but with those small differences. Maybe @WenfeiXiao Fiona can give more details. I guess Fiona may enter the estimated slope and intercept into the calculation with less digits when generating her YAML files.

**Comment by tillahoffmann** [comment]
Yeah, I think the + instead of - makes sense because that's what the scikit-learn linear regression uses internally.

**Comment by YWAN446** [comment]
Let's wait for Fiona to answer what could be the reason for those small differences in those values and then we can close this.

**Comment by WenfeiXiao** [comment]
I originally used young2020['concentration'] = np.where(young2020['Ctvalue'] == 38, 1, 10 ** (14.1147 - 0.3231 * young2020['Ctvalue'])). When I substituted the slope and intercept, sorry I forgot to remove the sign. A slight difference may have introduced due to preserving the last few digits. Apologies for the confusion.

**Review by tillahoffmann — APPROVED** [review]
That makes sense, thank you for the explanation!

**Pull request merged**